include/graph/graph.hpp

```diff
@@ -5,49 +5,18 @@
 #include <string>
 #include <vector>
 
-namespace itlab_2023 {
-
-enum LayerType {
-  kInput,
-  kPooling,
-  kNormalization,
-  kDropout,
-  kElementWise,
-  kConvolution,
-  kFullyConnected,
-  kOutput
-};
-
-class LayerExample {
- private:
-  int id_;
-  std::string name_;
-  LayerType type_;
-  std::string version_;
-  int numInputs_;
-  int numNeurons_;
-  std::vector<int> primer_;
+#include "layers/Layer.hpp"
 
- public:
-  LayerExample(LayerType type1) : type_(type1) {}
-  LayerType getType() { return type_; }
-  int getNumInputs() const { return numInputs_; }
-  int getNumNeurons() const { return numNeurons_; }
-  int checkID() const { return id_; }
-  void giveID(int id1) { id_ = id1; }
-  void In(const std::vector<int>& a) { primer_ = a; }
-  void Work() {}
-  std::vector<int> Out() { return primer_; }
-};
+namespace itlab_2023 {
 
 class Graph {
   int BiggestSize_;
   int V_;
-  std::vector<LayerExample> layers_;
+  std::vector<Layer*> layers_;
   std::vector<int> arrayV_;
   std::vector<int> arrayE_;
-  std::vector<int> startvec_;
-  std::vector<int>* outvector_;
+  Tensor startten_;
+  Tensor* outten_;
```

Comment:
What is startten in that context?

Reply:
in general, this is something like an input. But after each completed layer, it changes to be transferred to the next layer. So this is an input, but I wanted to call it something else.

Reply:
The name is really not clear and does not follow the naming strategy that we use in this class as well (camel case)

Reply:
fixed